src/dotnet-interactive-vscode/insiders/src/notebookControllers.ts

```diff
@@ -204,8 +202,21 @@ export async function updateCellLanguages(document: vscode.NotebookDocument): Pr
 }
 
 async function updateCellOutputs(executionTask: vscode.NotebookCellExecution, outputs: Array<vscodeLike.NotebookCellOutput>): Promise<void> {
-    const reshapedOutputs = outputs.map(o => new vscode.NotebookCellOutput(o.items.map(oi => generateVsCodeNotebookCellOutputItem(oi.data, oi.mime))));
-    await executionTask.replaceOutput(reshapedOutputs);
+    const previousOutput = executionTask.cell.outputs.length ? executionTask.cell.outputs[executionTask.cell.outputs.length - 1] : undefined;
+    const previousOutputItem = previousOutput?.items.length ? previousOutput.items[previousOutput.items.length - 1] : undefined;
+    await Promise.all(outputs.map(async (o, index) => {
+        const items = o.items.map(oi => generateVsCodeNotebookCellOutputItem(oi.data, oi.mime, oi.stream));
+        // If all of these items are of the same stream type & previous item is the same stream, then append it.
+        if (index === 0 && previousOutput && previousOutputItem?.mime && ['application/vnd.code.notebook.stderr', 'application/vnd.code.notebook.stdout'].includes(previousOutputItem?.mime)){
```

Comment:
If we get output from stdout or std err, then concatenate them (so that we have a single entry in generated output file)

Reply:
These mime types are documented in VS Code, hence safe to hardcode